source-mongodb/pull.go

```diff
@@ -416,7 +431,7 @@ func (c *capture) BackfillCollection(ctx context.Context, client *mongo.Client,
 			if diff, err := oplogTimeDifference(ctx, client); err != nil {
 				return fmt.Errorf("could not read oplog, access to oplog is necessary: %w", err)
 			} else {
-				// TODO: This is just spamming logs, but users won't see these logs. Should we
+				// TODO: Users won't see these logs until there is an error. Should we
```

Comment:
TBH I think we should elevate this check to just being a check that is done in `Validate`, and only errors if its a new enough version of MongoDB to query the retention period directly (I think that's a thing?) and that retention period is lower than some threshold. Then there could be a "skip retention sanity check" option like `source-mysql`. It could be up for debate how this would work on versions too old for that retention period to be queried directly - either always fail unless the "skip" option is checked, or always pass with a warning.

Mainly my thoughts here are motivated by seeing strange things in some of my own testing: Logs like "your retention period is 35 seconds" when I have loaded some test data, etc. It just doesn't seem practical to check for a sufficient oplog period in this way, altough I certainly understand there is precedent for it.

Reply:
I updated this, so now we first check the `oplogMinRetentionHours` configuration, if it is set, we require that it be more than 24 hours. If it is not set, we use the _approximation_ method to see if the oplog _seems_ small, we do let the user know though that this is an approximation.